src/util/submit-addon.js

```diff
@@ -90,6 +90,9 @@ export default class Client {
     userAgentString,
   }: ClientConstructorParams) {
     this.apiAuth = apiAuth;
+    if (!baseUrl.href.endsWith('/')) {
```

Comment:
If the URL somehow contains a `?` or `#` but not a `/`, then this would not work as desired.

This would fix the issue:

```js
if (!baseUrl.pathname.endsWith('/')) {`
  baseUrl = new URL(baseUrl);
  baseUrl.pathname += "/";
}
```

P.S. The query and ref will be stripped by `new URL('addons/', baseUrl)` anyway, so an alternative is to assert that `baseUrl.search` and `baseUrl.hash` are both empty.

Reply:
The fix suggestion sounds good.  I'm not sure it's worth the extra effort to assert on `baseUrl.search` and `baseUrl.hash` though - the code is too specialised to the AMO API here, so trying to support another API that has a different url style is out of scope. (My vote is just to ignore it and let them be stripped)

Reply:
https://github.com/mozilla/web-ext/pull/2621/commits/757f47a646311ab50966e64e49035dbd884734f7 is your fix(ish)